src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Col, Icon, Row, Stack,
+} from '@openedx/paragon';
+import { Calendar } from '@openedx/paragon/icons';
+import dayjs from 'dayjs';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { defineMessages, useIntl } from '@edx/frontend-platform/i18n';
+import { DATE_FORMAT } from '../data';
+import {
+  determineAllocatedCourseRunAssignmentsForCourse,
+  getSoonestEarliestPossibleExpirationData,
+  useCourseMetadata,
+  useRedeemablePolicies,
+} from '../../app/data';
+
+const messages = defineMessages({
+  importantDates: {
+    id: 'enterprise.course.about.page.important-dates.title',
+    defaultMessage: 'Important dates',
+    description: 'Title for the important dates section on the course about page',
+  },
+  enrollByDate: {
+    id: 'enterprise.course.about.page.important-dates.enroll-by-date',
+    defaultMessage: 'Enroll-by date',
+    description: 'Enroll-by date for the important dates section on the course about page',
+  },
+  courseStarts: {
+    id: 'enterprise.course.about.page.important-dates.course-starts',
+    defaultMessage: 'Course starts',
+    description: 'Course starts for the important dates section on the course about page in future tense',
+  },
+  courseStarted: {
+    id: 'enterprise.course.about.page.important-dates.course-started',
+    defaultMessage: 'Course started',
+    description: 'Course started the important dates section on the course about page in past tense',
+  },
+});
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+
+  if (!hasAssignedCourseRuns || (!!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(courseRunKey))) {
```

Comment:
Yep validated the behavior is now working as expected post refactor 👍🏽

Reply:
Ok re-validated and turns out we can't put a `!` in front of `hasCourseRunAssignmentForCourse` AND remove the `!` from `allocatedCourseRunAssignmentKeys` check for `courseRunKey` because it would invert the value of the truthy `!!courseRunKey` if the check were to short circuity on `courseRunKey`